eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1306 @@
+
+  if (regex == NULL)
+    return NULL;
+
+  return g_regex_replace (regex_compiled,
+                          content,
+                          -1,
+                          0,
+                          substitution,
+                          0,
+                          error);
+}
+
+static char *
+strip_body_tags (const char  *html,
+                 GError     **error)
+{
+  g_autofree char *stripped_start_tags = regex_substitute ("^\\s*<html>\\s*<body>",
+                                                            "",
+                                                            html,
+                                                            error);
+
+  if (stripped_start_tags == NULL)
+    return NULL;
+
+  return regex_substitute ("<\\/body>\\s*<\\/html>\\s*$",
+                           "",
+                           stripped_start_tags,
+                           error);
+}
+
+static GFile *
+template_file (const char *filename)
+{
+  g_autofree char *uri = g_strdup_printf ("resource:///com/endlessm/knowledge/data/templates/%s", filename);
+  return g_file_new_for_uri (uri);
+}
+
+static char *
+_renderer_render_legacy_content (EknrRenderer *renderer,
+                                 const char   *body_html,
+                                 const char   *source,
+                                 const char   *source_name,
+                                 const char   *original_uri,
+                                 const char   *license,
+                                 const char   *title,
+                                 gboolean      show_title,
+                                 gboolean      use_scroll_manager,
+                                 GError      **error)
+{
+  g_autoptr(GFile) file = template_file ("legacy-article.mst");
+  g_autofree char *stripped_body = strip_body_tags (body_html, error);
+  GVariantDict vardict;
+  g_autoptr(GVariant) variant = NULL;
+  GVariant *disclaimer = NULL; /* floating */
+
+  if (stripped_body == NULL)
+    return NULL;
+
+  disclaimer = get_legacy_disclaimer_section_content (source,
+                                                      source_name,
+                                                      original_uri,
+                                                      license,
+                                                      title,
+                                                      error);
+
+  if (disclaimer == NULL)
+    return NULL;
+
+  g_variant_dict_init (&vardict, NULL);
+
+  g_variant_dict_insert_value (&vardict,
+                               "title",
+                               show_title ? g_variant_new_string (title) : g_variant_new_boolean (FALSE));
+  g_variant_dict_insert_value (&vardict, "body-html", g_variant_new_take_string (g_steal_pointer (&stripped_body)));
+  g_variant_dict_insert_value (&vardict, "disclaimer", disclaimer);
+  g_variant_dict_insert_value (&vardict, "copy-button-text", g_variant_new_string (_("Copy")));
```

Comment:
Since this library requires internationalization unlike libdmodel, this will require adding gettext machinery to the build files, moving the translations over to this repo from eos-knowledge-lib, and creating a Transifex project. (But see the main review comment, it might be possible to avoid that.)

Reply:
Still needs addressing. I also realized there are translatable strings in the legacy template as well, so avoiding gettext will not be possible even if we remove the non-legacy template.

Reply:
Okay, I'll try to get this done tonight